lib/daemon.js

```diff
@@ -68,21 +49,16 @@ class Daemon {
     Object.keys(this._pollers).forEach(pollerId => this._removePoller(pollerId))
   }
 
-  _addPoller (descriptor, forcePoll = false) {
-    this._logger.info('spinning up poller', descriptor)
-
-    const poller = new Poller({
-      backends: this._backends,
-      intervalMilliseconds: this._pollerIntervalMilliseconds,
-      kubeClient: this._kubeClient,
-      logger: this._logger,
-      metrics: this._metrics,
-      namespace: descriptor.namespace,
-      secretDescriptor: descriptor.secretDescriptor,
-      ownerReference: descriptor.ownerReference
-    })
-
-    this._pollers[descriptor.id] = poller.start({ forcePoll })
+  _addPoller (descriptor) {
+    this._logger.info('spinning up poller for', descriptor.name, 'in', descriptor.namespace)
+
+    if (this._pollers[descriptor.id]) {
+      throw new Error(`Poller for ${descriptor.id} already exists`)
+    }
+
+    const poller = this._pollerFactory.createPoller(descriptor)
```

Comment:
I'm a little confused at the types here: `descriptor` is result of `_createPollerDescriptor`, but `.createPoller` takes an `ExternalSecret` manifest / object.  Is there an actual type problem, or confusion with variable naming?

Reply:
Hmm well the full externalSecret now exists in in the descriptor, just destructuring externalSecret from the descriptor as nothing more is needed to create the poller, any suggestions to make it more clear?

Ie `this._pollerFactory.createPoller({ externalSecret: descriptor.externalSecret} )`

Reply:
doh, i missed the destructuring in the `.createPoller`.  ignore me :)